geomstats/geometry/siegel.py

```diff
@@ -231,7 +228,7 @@ def inner_product(self, tangent_vec_a, tangent_vec_b, base_point):
         inner_product : array-like, shape=[...,]
             Inner-product.
         """
-        identity = _create_identity_mat(base_point.shape, dtype=base_point.dtype)
+        identity = _eye_like(base_point)
```

Comment:
looking back to the code, I've realized that every time we use this function, we actually only need to create a (n, n) identity matrix with same type as `base_point`. Therefore, I suggest we remove the function `_eye_like` and do:

`identity = gs.eye(space.shape, dtype=base.dtype)`

I think it will work fine gpu.

Reply:
Hello @luisfpereira,

No, using: 
`identity = gs.eye(space.shape, dtype=base.dtype)`
is not sufficient for two reasons.

The first reason is the shape.
The first input of the function torch.eye should be an integer `n`, not a shape.
Then, a 2D tensor is returned.
I would like my identity matrix to be 2D or 3D (in this case it is a repetition of 2D identity matrices along the first axis), depending on the shape of the input arrays of my function.

The second reason is the device.
When using the PyTorch backend with GPUs enable, the following code:

```
a = torch.tensor([[0.5 + 0.2j, 0], [0, 0.5]], requires_grad=True, device=device)
```
returns
```
tensor([[0.5000+0.2000j, 0.0000+0.0000j],
        [0.0000+0.0000j, 0.5000+0.0000j]], device='cuda:0', requires_grad=True)
```

Then
```
gs.eye(a.shape[0], dtype=a.dtype).device
```
returns
```
device(type='cpu')
```
which is not what we want.

However, 
```
gs.eye(a.shape[0], dtype=a.dtype, device=a.device).device
```
returns
```
device(type='cuda', index=0)
```
which is what we want.

So it is not sufficient to use the `dtype` optional input argument, we have to use the `device` optional input argument.
However, note that `np.eye` do not accept a `device` argument.

Reply:
It think that the easiest way to do what I want is to use `_eye_like` as I have proposed, using the function `gs.zeros_like` which is designed to create a zero matrix compatible with the input matrix for both the `NumPy` and the `PyTorch` backend.

Reply:
For the shapes, even though it looks like you need to create 3d arrays, actually it suffices (and is better) to create 2d arrays, as the operations where they are used that advantage of broadcasting (thus, we save some memory).

For the device, I'm not suggesting passing `device` as input, I'm suggesting that when we set globally the device to be the GPU, then any tensor is created by default on the GPU. I thought this was actually the behavior of `pytorch`. But if not, we can easily implement this behavior in the backend, only concerning pytorch.
This is the part that we need to test, probably not the full code.